e2e/cypress/integration/messaging/message_reply_input_box_expand_spec.js

```diff
@@ -0,0 +1,103 @@
+// ***************************************************************
+
+import * as TIMEOUTS from '../../fixtures/timeouts';
+
+describe('Messaging', () => {
+    before(() => {
+        // # Login and navigate to town-square
+        cy.toMainChannelView('user-1');
+
+        // # Post a new message to ensure there will be a post to click on
+        cy.postMessage('Hello ' + Date.now());
+    });
+
+    it('M18706-Input box on reply thread can expand', () => {
+        const maxReplyCount = 15;
+        const halfViewportHeight = Cypress.config('viewportHeight') / 2;
+        const padding = 4;
+        const postCreateContainerDefaultHeight = 184 - padding;
+        const replyTextBoxDefaultHeight = 100;
+        const postCreateContainerClassName = 'post-create__container';
+        const replyTextBoxId = 'reply_textbox';
+        let newLinesCount;
+        let newLines;
+
+        // # Click "Reply"
+        cy.getLastPostId().then((postId) => {
+            cy.clickPostCommentIcon(postId);
+        });
+
+        // # Post several replies and verify last reply
+        cy.get(`#${replyTextBoxId}`).clear().should('be.visible').as('replyTextBox');
+        for (let i = 1; i <= maxReplyCount; i++) {
+            cy.get('@replyTextBox').type(`post ${i}`).type('{enter}');
+        }
+        verifyLastReply(maxReplyCount);
+
+        // # Get post create container and reply text box
+        cy.document().then((doc) => {
+            const postCreateContainer = doc.getElementsByClassName(postCreateContainerClassName)[0];
+            const replyTextBox = doc.getElementById(replyTextBoxId);
+
+            // * Check if post create container has default offset height and less than 50% of viewport height
+            expect(postCreateContainer.offsetHeight - padding).to.eq(postCreateContainerDefaultHeight).and.lessThan(halfViewportHeight);
+
+            // * Check if reply text box has default offset height and less than post create container default offset height
+            expect(replyTextBox.offsetHeight).to.eq(replyTextBoxDefaultHeight).and.lessThan(postCreateContainerDefaultHeight);
+        });
+
+        // # Enter new lines into RHS so that box should reach max height, verify last reply, and verify heights
+        newLinesCount = 25;
+        newLines = '{shift}{enter}'.repeat(newLinesCount);
+        cy.get('@replyTextBox').type(newLines);
+        verifyLastReply(maxReplyCount);
+        verifyHeights(postCreateContainerClassName, replyTextBoxId, padding, halfViewportHeight, postCreateContainerDefaultHeight);
+
+        // # Enter more new lines into RHS, verify last reply, and verify heights
+        newLinesCount *= 2;
+        newLines = '{shift}{enter}'.repeat(newLinesCount);
+        cy.get('@replyTextBox').type(newLines);
+        verifyLastReply(maxReplyCount);
+        verifyHeights(postCreateContainerClassName, replyTextBoxId, padding, halfViewportHeight, postCreateContainerDefaultHeight);
```

Comment:
Since you're entering new lines and verifying last reply and height of the reply box, maybe condense these lines in a single function?

Reply:
Updated in latest commit